test/unit/test_layer.py

```diff
@@ -348,3 +348,34 @@ def test_localNetUpSampleResnetBlock():
     assert isinstance(model._additive_upsampling, layer.AdditiveUpSampling)
     assert isinstance(model._conv3d_block, layer.Conv3dBlock)
     assert isinstance(model._residual_block, layer.LocalNetResidual3dBlock)
+
+
+def test_BSplines3DTransform():
+    """
```

Comment:
Here you are testing multiple different cases, please split tests into diff funcs inside a big `TestBSplines3DTransform`

check the doc/example here

https://deepreg.readthedocs.io/en/latest/contributing/test.html